### ISSUE TITLE
Introduce ModelCard entity to document ML model purpose, data provenance & responsible-AI metadata

### ISSUE THREAD

**Issue by harshach** [open-metadata/OpenMetadata#22134]
Modern ML practice increasingly demands transparency, governance, and regulatory readiness. Frameworks such as Google’s Model Cards and Hugging Face’s model-card metadata formalize what must be disclosed about every production model—its intended use, datasets, evaluation metrics, ethical risks, and limitations.

OpenMetadata already captures technical lineage for MlModel (features, hyper-params, store, serving), but it does not store the narrative & compliance-oriented metadata required for audits, consumer trust, or internal risk reviews. Today that information lives in scattered wikis, slide decks, or the HF hub—outside the unified metadata graph—making it hard to:

- answer “Which datasets trained model X?” or “Where are the bias caveats for model Y?”
- audit models against emerging AI regulations (EU AI Act, US NIST RMF, etc.)
- surface responsible-AI warnings in the same UI where analysts explore data lineage.

## Tasks
- Add modelCard.json schema (see PR template below)
- Add modelCard reference field to MlModel schema
- REST API endpoints: POST /modelCards, GET /modelCards/{id}, PATCH
-  UI: “Model Card” entity page

## Model Card Specification

```
{
  "$id": "https://open-metadata.org/schema/entity/data/modelCard.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "ModelCard",
  "$comment": "@om-entity-type",
  "description": "A ModelCard stores the documentation-oriented metadata of an ML model—its purpose, data provenance, evaluation metrics, ethical considerations and limitations. It complements (but is separate from) the runtime-oriented MlModel entity.",
  "type": "object",
  "javaType": "org.openmetadata.schema.entity.data.ModelCard",
  "javaInterfaces": ["org.openmetadata.schema.EntityInterface"],
  "definitions": {
    "datasetReference": {
      "title": "DatasetReference",
      "type": "array",
       "$ref": "../../type/entityReference.json"
      }
    },
    "metric": {
      "title": "Metric",
      "type": "object",
      "javaType": "org.openmetadata.schema.type.ModelMetric",
      "additionalProperties": false,
      "required": ["name", "value"],
      "properties": {
        "name":   { "type": "string", "description": "Metric name (e.g. accuracy, F1)." },
        "value":  { "type": "number", "description": "Numeric value of the metric." },
        "unit":   { "type": "string", "description": "Unit or scale (%, score, etc.).", "default": null },
        "slice":  { "type": "string", "description": "Data slice / condition (e.g. age<18).", "default": null },
        "description": {
          "$ref": "../../type/basic.json#/definitions/markdown",
          "description": "Free-text explanation of the metric calculation.",
          "default": null
        }
      }
    },
  "properties": {
    "id":                    { "$ref": "../../type/basic.json#/definitions/uuid" },
    "name":                  { "$ref": "../../type/basic.json#/definitions/entityName" },
    "fullyQualifiedName":    { "$ref": "../../type/basic.json#/definitions/fullyQualifiedEntityName" },
    "displayName":           { "type": "string", "description": "Display name of the card.", "default": null },
    "description":           { "$ref": "../../type/basic.json#/definitions/markdown" },
    "mlModel": {
      "$ref": "../../type/entityReference.json",
      "description": "Reference to the MlModel this card describes."
    },
    "modelType":     { "type": "string", "description": "Problem category (e.g. text-classification)." },
    "architecture":  { "type": "string", "description": "Model family / architecture (e.g. Transformer-BERT)." },
    "modelVersion":  { "type": "string", "description": "Version or artefact hash." },
    "license":       { "type": "string", "description": "Usage licence (Apache-2.0, MIT, proprietary, …)." },
    "sourceUrl":     { "$ref": "../../type/basic.json#/definitions/sourceUrl" },
    "intendedUse":   { "$ref": "../../type/basic.json#/definitions/markdown" },
    "outOfScopeUse": { "$ref": "../../type/basic.json#/definitions/markdown" },
    "trainingData": {
      "type": "array",
      "items": { "$ref": "#/definitions/datasetReference" },
      "description": "Datasets used to train the model.",
      "default": null
    },
    "evaluationData": {
      "type": "array",
      "items": { "$ref": "#/definitions/datasetReference" },
      "description": "Datasets used to validate/test the model.",
      "default": null
    },
    "metrics": {
      "type": "array",
      "items": { "$ref": "#/definitions/metric" },
      "description": "Performance metrics (optionally per slice).",
      "default": null
    },
    "relevantFactors": {
      "type": "array",
      "items": { "type": "string" },
      "description": "Factors known to affect performance.",
      "default": null
    },
    "ethicalConsiderations": { "$ref": "../../type/basic.json#/definitions/markdown", "default": null },
    "limitations":           { "$ref": "../../type/basic.json#/definitions/markdown", "default": null },
    "computeResources": { "type": "string", "description": "Hardware/accelerator summary.", "default": null },
    "hyperparameters": {
      "type": "array",
      "items": { "$ref": "../mlmodel.json#/definitions/hyperParameter" },
      "description": "Key hyper-parameters.",
      "default": null
    },
    "usageExample": {"$ref": "../../type/basic.json#/definitions/markdown", "default": null }
    "href":               { "$ref": "../../type/basic.json#/definitions/href" },
    "version":            { "$ref": "../../type/entityHistory.json#/definitions/entityVersion" },
    "updatedAt":          { "$ref": "../../type/basic.json#/definitions/timestamp" },
    "updatedBy":          { "type": "string" },
    "service":            { "$ref": "../../type/entityReference.json" },
    "serviceType":        { "type": "string", "description": "Parent service type.", "default": null },
    "owners":             { "$ref": "../../type/entityReferenceList.json" },
    "followers":          { "$ref": "../../type/entityReferenceList.json" },
    "tags":               { "type": "array", "items": { "$ref": "../../type/tagLabel.json" }, "default": [] },
    "changeDescription":  { "$ref": "../../type/entityHistory.json#/definitions/changeDescription", "default": null },
    "incrementalChangeDescription": {
      "$ref": "../../type/entityHistory.json#/definitions/changeDescription",
      "default": null
    },
    "deleted":            { "type": "boolean", "default": false },
    "extension":          { "$ref": "../../type/basic.json#/definitions/entityExtension", "default": null },
    "sourceUrl":          { "$ref": "../../type/basic.json#/definitions/sourceUrl", "default": null },
    "domain":             { "$ref": "../../type/entityReference.json", "default": null },
    "votes":              { "$ref": "../../type/votes.json", "default": null },
    "lifeCycle":          { "$ref": "../../type/lifeCycle.json", "default": null },
    "certification":      { "$ref": "../../type/assetCertification.json", "default": null }
  }
}
```

**Comment by pmbrull** [comment]
Awesome!! Few notes:

1. Should we reuse the [mlHyperParameter](https://github.com/open-metadata/OpenMetadata/blob/main/openmetadata-spec/src/main/resources/json/schema/entity/data/mlmodel.json#L116) definition from the ML Model Entity directly? (Maybe it was already copied from there and just added here as a ref, in that case perfect)
    1. In this line of thought, should we keep the Hyper Parameters directly on the ML Model? Should we move them from the ML Model entity to the Model Card as a whole?
3.  On the `metric` definition we are adding a `dataset` prop as    
 
    ```
    "dataset": {
              "$ref": "../../type/entityReference.json",
            },
    ```
    I am not sure this is necessary to be there, since these metrics would be computed on the `evaluationData` defined below.
3. IMO `trainingData` and `evaluationData` could be more generic, not necessarily an `EntityReference`. I'd keep these as `Markdown` to let ppl explain (and link if needed) these pieces. Train of thought is that you might not always be training on data that would appear in a catalog. E.g., doing a Computer Vision project to extract passport documentation, you'd just get all passport images your ops teams have been manually checking, but you'd rarely surface these to OpenMetadata since that's not a "curated" dataset of any kind, rather a step on an ops process. Another example could be fine-tuning an LLM, where while your training data could be kept in-house, you might want to evaluate against LLM Leaderboards.
4. Most models in HF have examples on how to use them. Do we want to keep just the `description` and let people inform that there, or do we want to be more explicit as when we defined the Metric Entity?

**Comment by harshach** [comment]
1. Yes
2. Ya that needs to be fixed
3. my intent of training data and evaluation data is to link to existing assets such as S3 buckets, or tables themseleves
4. I would like to be explicit

**Comment by pmbrull** [comment]
ok thanks, I'm updating the ticket to reflect these. For (3), let's keep it as-is and see how people use it. We always have room to make it a string (vs the other way around)

UPDATE: edited the config with the pieces to remove. Added `usageExample` as Markdown to let people explain the steps and put code freely
